src/app/users/login.ts

```diff
@@ -55,25 +59,30 @@ export const login = async ({
   await rewardFailedLoginAttemptPerIpLimits(ip)
   await rewardFailedLoginAttemptPerPhoneLimits(phone)
 
-  const userRepo = UsersRepository()
```

Comment:
Nit, adding a suggestion that could be more intuitive as we separate the `User` and `Account` concerns in persistence layer.

```ts
  const user = await UsersRepository().findByPhone(phone)
  if (user instanceof Error && !(user instanceof CouldNotFindUserFromPhoneError)) {
    return user
  }

  if (user instanceof CouldNotFindUserFromPhoneError) {
    subLogger.info({ phone }, "new user signup")
    const userRaw: NewUserInfo = { phone }
    const account = await createAccountForPhoneSchema({
      newUserInfo: userRaw,
      config: getDefaultAccountsConfig(),
    })
    if (account instanceof Error) return account

    return createToken({ uid: account.id, network })
  }

  subLogger.info({ phone }, "user login")
  const account = await AccountsRepository().findByUserId(user.id)
  if (account instanceof Error) return account

  return createToken({ uid: account.id, network })
}

```